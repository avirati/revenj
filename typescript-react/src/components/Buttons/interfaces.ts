export interface ISimpleButton {
  className?: string;
  disabled?: boolean;
  label: string;
  url?: string;
  isExternalUrl?: boolean;
  tooltip?: string;
  onClick?: () => void;
}

export interface IComponentButton {
  Component: React.ComponentType<{}>;
}

export type ButtonItem = ISimpleButton | IComponentButton;

export const isComponentButton = (it: any): it is IComponentButton =>
  it != null && (it as IComponentButton).Component != null;
