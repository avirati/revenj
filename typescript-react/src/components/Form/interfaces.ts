export enum FormType {
  View = 'View',
  Create = 'Create',
  Edit = 'Edit',
}

export interface IFormError {
  message: string | JSX.Element;
  code?: string | number;
}

export enum ErrorType {
  Error,
  Warning,
  QAPink,
}
