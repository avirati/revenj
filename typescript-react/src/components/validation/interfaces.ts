import { Numeric } from '../../util/NumberUtils/NumberUtils';

// Operator argument types
export type Requireable = any;
export type Ord = Numeric;
export type Num = Numeric;
export type Eq = any;

// Operator interfaces
export interface IOperator<B, I> {
  (baseValue: B): (inputValue: I) => boolean;
}
export interface IRequireableOperator extends IOperator<undefined, Requireable> {
  (): (inputValue: Requireable) => boolean;
}

export interface IOrdOperator extends IOperator<Ord, Ord> {
  (baseValue: Numeric): (inputValue: Numeric) => boolean;
}

export interface IOrdAlphaOperator extends IOperator<string, string> {}

export interface INumOperator extends IOperator<Ord, Ord> {
  (baseValue: Numeric): (inputValue: Numeric) => boolean;
}

export interface IEqOperator extends IOperator<any, any> {
  <T>(baseValue: T): (inputValue: T) => boolean;
}

export interface IGetValidatorBaseValue<B, F, P> {
  (allValues: F, componentProps: P, fieldName: string): B;
}

export interface IGetValidatorErrorMessage<B, I, F, P> {
  (validationBaseValue: B, inputValue: I, allValues: F, componentProps: P, fieldName: string): string;
}

export interface IParseValue<R, V> {
  (input: R): V;
}

export type ValidationError = string | undefined;
export type Validator<I, F, P> = (inputValue: I, allValues?: F, props?: P, name?: keyof F) => ValidationError;

export enum ValidateTrigger {
  Always = 'Always',
  IfInputIsSet = 'IfInputIsSet',
  IfBaseIsSet = 'IfBaseIsSet',
  IfInputAndBaseAreSet = 'IfInputAndBaseAreSet',
}

interface IValidatorCreatorProps<B, I, F, P> {
  validateOverride?: ValidateTrigger;
  getValidatorBaseValueOverride?: IGetValidatorBaseValue<B, F, P> | B;
  getValidatorErrorMessageOverride?: IGetValidatorErrorMessage<B, I, F, P> | string;
  getValueFromInputOverride?: IParseValue<any, I>;
  getValueFromBaseOverride?: IParseValue<any, B>;
}

export type ValidatorCreator<B, I, F, P> = (props?: IValidatorCreatorProps<B, I, F, P>) => Validator<I, F, P>;

export type Operator<B, I> = (baseValue: B) => (inputValue: I) => boolean;

export interface IFormFieldSpecification {
  label: string;
  path: string;
}

// In many cases, the path is not actually required
export interface IFormFieldSpecificationDescriptor extends Omit<IFormFieldSpecification, 'path'> {
  path?: IFormFieldSpecification['path'];
}

export interface IFormFieldValue<T> extends IFormFieldSpecification {
  value: T;
}

export interface ITransformer<V, R> {
  (value: V): R;
}

export type RegexOperator = (regex: RegExp) => (inputValue: string) => boolean;

export type Preprocessor<T> = (value: T) => T;
