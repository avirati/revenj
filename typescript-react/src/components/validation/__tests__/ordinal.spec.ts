import { BigNumber } from 'bignumber.js';

import { Ord } from '../interfaces';
import {
  gteOrdAlphaCreator,
  gteOrdCreator,
  gtOrdAlphaCreator,
  gtOrdCreator,
  inIntervalCreator,
  inSegmentCreator,
  lteOrdAlphaCreator,
  lteOrdCreator,
  ltOrdAlphaCreator,
  ltOrdCreator,
} from '../ordinal';

describe('validation', () => {
  describe('gtOrdCreator', () => {
    const gt10 = gtOrdCreator(10)();
    const gt100 = gtOrdCreator('100')();
    const gt10Custom = gtOrdCreator(10)({
      getValidatorErrorMessageOverride: (value) => `This quick brown fox can only jump over more than ${value} lazy dogs.`,
    });
    const gt100Custom = gtOrdCreator('100')({
      getValidatorErrorMessageOverride: (value) => `This makes no sense! Be greater than ${value}!`,
    });

    it('should return an error message on value that is less than or equal', () => {
      expect(gt10(9, {})).toBeDefined();
      expect(gt10(9, {})).toMatchSnapshot();
      expect(gt10(10, {})).toBeDefined();
      expect(gt10(10, {})).toMatchSnapshot();
      expect(gt100('99', {})).toBeDefined();
      expect(gt100('99', {})).toMatchSnapshot();
      expect(gt100('100', {})).toBeDefined();
      expect(gt100('100', {})).toMatchSnapshot();
    });

    it('should return undefiend on value that is greater', () => {
      expect(gt10(11, {})).toBeUndefined();
      expect(gt100('1000', {})).toBeUndefined();
    });

    it('should return a custom error message on failure when value is not greater than', () => {
      expect(gt10Custom(10, {})).toBeDefined();
      expect(gt10Custom(10, {})).toMatchSnapshot();
      expect(gt10Custom(9, {})).toBeDefined();
      expect(gt10Custom(9, {})).toMatchSnapshot();
      expect(gt100Custom('99', {})).toBeDefined();
      expect(gt100Custom('99', {})).toMatchSnapshot();
      expect(gt100Custom('100', {})).toBeDefined();
      expect(gt100Custom('100', {})).toMatchSnapshot();
    });
  });

  describe('gteOrdCreator', () => {
    const gte10 = gteOrdCreator(10)();
    const gte100 = gteOrdCreator('100')();
    const gte10Custom = gteOrdCreator(10)({
      getValidatorErrorMessageOverride: (value) => `This quick brown fox can only jump over ${value} or more lazy dogs.`,
    });
    const gte100Custom = gtOrdCreator('100')({
      getValidatorErrorMessageOverride: (value) => `This makes no sense! Be greater than or equal to ${value}!`,
    });

    it('should return an error message on value that is less than', () => {
      expect(gte10(9, {})).toBeDefined();
      expect(gte10(9, {})).toMatchSnapshot();
      expect(gte100('99', {})).toBeDefined();
      expect(gte100('99', {})).toMatchSnapshot();
    });

    it('should return undefiend on value that is greater than or equal', () => {
      expect(gte10(11, {})).toBeUndefined();
      expect(gte10(10, {})).toBeUndefined();
      expect(gte100('100', {})).toBeUndefined();
      expect(gte100('1000', {})).toBeUndefined();
    });

    it('should return a custom error message on failure when value is not greater than or equal', () => {
      expect(gte10Custom(9, {})).toBeDefined();
      expect(gte10Custom(9, {})).toMatchSnapshot();
      expect(gte100Custom('99', {})).toBeDefined();
      expect(gte100Custom('99', {})).toMatchSnapshot();
    });
  });

  describe('ltOrdCreator', () => {
    const lt10 = ltOrdCreator(10)();
    const lt100 = ltOrdCreator('100')();
    const lt10Custom = ltOrdCreator(10)({
      getValidatorErrorMessageOverride: (value) => `This quick brown fox can only jump under ${value} lazy dogs.`,
    });
    const lt100Custom = ltOrdCreator('100')({
      getValidatorErrorMessageOverride: (value) => `This makes no sense! Be less than ${value}!`,
    });

    it('should return an error message on value that is greater than or equal', () => {
      expect(lt10(11, {})).toBeDefined();
      expect(lt10(11, {})).toMatchSnapshot();
      expect(lt10(10, {})).toBeDefined();
      expect(lt10(10, {})).toMatchSnapshot();
      expect(lt100('100', {})).toBeDefined();
      expect(lt100('100', {})).toMatchSnapshot();
      expect(lt100('1000', {})).toBeDefined();
      expect(lt100('1000', {})).toMatchSnapshot();
    });

    it('should return undefiend on value that is less than', () => {
      expect(lt10(9, {})).toBeUndefined();
      expect(lt100('99', {})).toBeUndefined();
    });

    it('should return a custom error message on failure when value is not less than', () => {
      expect(lt10Custom(10, {})).toBeDefined();
      expect(lt10Custom(10, {})).toMatchSnapshot();
      expect(lt10Custom(11, {})).toBeDefined();
      expect(lt10Custom(11, {})).toMatchSnapshot();
      expect(lt100Custom('100', {})).toBeDefined();
      expect(lt100Custom('100', {})).toMatchSnapshot();
      expect(lt100Custom('1000', {})).toBeDefined();
      expect(lt100Custom('1000', {})).toMatchSnapshot();
    });
  });

  describe('lteOrdCreator', () => {
    const lte10 = lteOrdCreator(10)();
    const lte100 = lteOrdCreator('100')();
    const lte10Custom = lteOrdCreator(10)({
      getValidatorErrorMessageOverride: (value) => `This quick brown fox can only jump under ${value} or less lazy dogs.`,
    });
    const lte100Custom = ltOrdCreator('100')({
      getValidatorErrorMessageOverride: (value) => `This makes no sense! Be less than or equal to ${value}!`,
    });

    it('should return an error message on value that is not less than or equal', () => {
      expect(lte10(11, {})).toBeDefined();
      expect(lte10(11, {})).toMatchSnapshot();
      expect(lte100('1000', {})).toBeDefined();
      expect(lte100('1000', {})).toMatchSnapshot();
    });

    it('should return undefiend on value that is less than or equal', () => {
      expect(lte10(9, {})).toBeUndefined();
      expect(lte10(10, {})).toBeUndefined();
      expect(lte100('100', {})).toBeUndefined();
      expect(lte100('99', {})).toBeUndefined();
    });

    it('should return a custom error message on failure when value is not less than or equal', () => {
      expect(lte10Custom(11, {})).toBeDefined();
      expect(lte10Custom(11, {})).toMatchSnapshot();
      expect(lte100Custom('1000', {})).toBeDefined();
      expect(lte100Custom('1000', {})).toMatchSnapshot();
    });
  });

  describe('gtOrdAlphaCreator', () => {
    const gtBBB = gtOrdAlphaCreator('BBB')();
    const gtBBBCustom = gtOrdAlphaCreator('BBB')({
      getValidatorErrorMessageOverride: (value) => `This makes no sense! Be greater than ${value}!`,
    });

    it('should return an error message on value that is less than or equal', () => {
      expect(gtBBB('AAA', {})).toBeDefined();
      expect(gtBBB('AAA', {})).toMatchSnapshot();
      expect(gtBBB('BBB', {})).toBeDefined();
      expect(gtBBB('BBB', {})).toMatchSnapshot();
    });

    it('should return undefiend on value that is greater', () => {
      expect(gtBBB('CCC', {})).toBeUndefined();
    });

    it('should return a custom error message on failure when value is not greater than', () => {
      expect(gtBBBCustom('BBB', {})).toBeDefined();
      expect(gtBBBCustom('BBB', {})).toMatchSnapshot();
      expect(gtBBBCustom('AAA', {})).toBeDefined();
      expect(gtBBBCustom('AAA', {})).toMatchSnapshot();
    });
  });

  describe('gteOrdAlphaCreator', () => {
    const gteBBB = gteOrdAlphaCreator('BBB')();
    const gteBBBCustom = gtOrdAlphaCreator('BBB')({
      getValidatorErrorMessageOverride: (value) => `This makes no sense! Be greater than or equal to ${value}!`,
    });

    it('should return an error message on value that is less than', () => {
      expect(gteBBB('AAA', {})).toBeDefined();
      expect(gteBBB('AAA', {})).toMatchSnapshot();
    });

    it('should return undefiend on value that is greater than or equal', () => {
      expect(gteBBB('BBB', {})).toBeUndefined();
      expect(gteBBB('CCC', {})).toBeUndefined();
    });

    it('should return a custom error message on failure when value is not greater than or equal', () => {
      expect(gteBBBCustom('AAA', {})).toBeDefined();
      expect(gteBBBCustom('AAA', {})).toMatchSnapshot();
    });
  });

  describe('ltOrdAlphaCreator', () => {
    const ltBBB = ltOrdAlphaCreator('BBB')();
    const ltBBBCustom = ltOrdAlphaCreator('BBB')({
      getValidatorErrorMessageOverride: (value) => `This makes no sense! Be less than ${value}!`,
    });

    it('should return an error message on value that is greater than or equal', () => {
      expect(ltBBB('CCC', {})).toBeDefined();
      expect(ltBBB('CCC', {})).toMatchSnapshot();
      expect(ltBBB('BBB', {})).toBeDefined();
      expect(ltBBB('BBB', {})).toMatchSnapshot();
    });

    it('should return undefiend on value that is less than', () => {
      expect(ltBBB('AAA', {})).toBeUndefined();
    });

    it('should return a custom error message on failure when value is not less than', () => {
      expect(ltBBBCustom('BBB', {})).toBeDefined();
      expect(ltBBBCustom('BBB', {})).toMatchSnapshot();
      expect(ltBBBCustom('CCC', {})).toBeDefined();
      expect(ltBBBCustom('CCC', {})).toMatchSnapshot();
    });
  });

  describe('lteOrdAlphaCreator', () => {
    const lteBBB = lteOrdAlphaCreator('BBB')();
    const lteBBBCustom = ltOrdAlphaCreator('BBB')({
      getValidatorErrorMessageOverride: (value) => `This makes no sense! Be less than or equal to ${value}!`,
    });

    it('should return an error message on value that is not less than or equal', () => {
      expect(lteBBB('CCC', {})).toBeDefined();
      expect(lteBBB('CCC', {})).toMatchSnapshot();
    });

    it('should return undefiend on value that is less than or equal', () => {
      expect(lteBBB('AAA', {})).toBeUndefined();
      expect(lteBBB('BBB', {})).toBeUndefined();
    });

    it('should return a custom error message on failure when value is not less than or equal', () => {
      expect(lteBBBCustom('CCC', {})).toBeDefined();
      expect(lteBBBCustom('CCC', {})).toMatchSnapshot();
    });
  });

  describe('inInterval', () => {
    const inInterval = inIntervalCreator({ min: 10, max: 20 })();

    it('should return undefined if value is inside the interval', () => {
      expect(inInterval(11, {})).toBeUndefined();
      expect(inInterval(19, {})).toBeUndefined();
      expect(inInterval('11', {})).toBeUndefined();
      expect(inInterval('19', {})).toBeUndefined();
      expect(inInterval(new BigNumber(11), {})).toBeUndefined();
      expect(inInterval(new BigNumber(19), {})).toBeUndefined();
    });

    it('should return undefined if any of the interval values are missing', () => {
      const missingMin = inIntervalCreator({ max: 10 })();
      const missingMax = inIntervalCreator({ min: 10 })();
      const missingBoth = inIntervalCreator({})();

      expect(missingMin(11, {})).toBeUndefined();
      expect(missingMax(9, {})).toBeUndefined();
      expect(missingBoth(1234, {})).toBeUndefined();
    });

    it('should return a validation error message is not inside the interval', () => {
      expect(inInterval(10, {})).toBeDefined();
      expect(inInterval(10, {})).toMatchSnapshot();
      expect(inInterval(20, {})).toBeDefined();
      expect(inInterval(20, {})).toMatchSnapshot();
      expect(inInterval('10', {})).toBeDefined();
      expect(inInterval('10', {})).toMatchSnapshot();
      expect(inInterval('20', {})).toBeDefined();
      expect(inInterval('20', {})).toMatchSnapshot();
      expect(inInterval(new BigNumber(10), {})).toBeDefined();
      expect(inInterval(new BigNumber(10), {})).toMatchSnapshot();
      expect(inInterval(new BigNumber(20), {})).toBeDefined();
      expect(inInterval(new BigNumber(20), {})).toMatchSnapshot();
    });

    it('should throw if range values or input supplied are not treatable as numbers', () => {
      const fail = (value: Ord) => () => {
        const failidator = inIntervalCreator({ min: 'not a number', max: 'somthing' })();
        failidator(value, {});
      };

      expect(fail(15)).toThrowErrorMatchingSnapshot();
      expect(fail('fifteen')).toThrowErrorMatchingSnapshot();
    });
  });

  describe('inSegment', () => {
    const inSegment = inSegmentCreator({ min: 10, max: 20 })();

    it('should return undefined if value is inside the segment', () => {
      expect(inSegment(10, {})).toBeUndefined();
      expect(inSegment(20, {})).toBeUndefined();
      expect(inSegment('10', {})).toBeUndefined();
      expect(inSegment('20', {})).toBeUndefined();
      expect(inSegment(new BigNumber(10), {})).toBeUndefined();
      expect(inSegment(new BigNumber(20), {})).toBeUndefined();
    });

    it('should return undefined if any of the segment values are missing', () => {
      const missingMin = inSegmentCreator({ max: 10 })();
      const missingMax = inSegmentCreator({ min: 10 })();
      const missingBoth = inSegmentCreator({})();

      expect(missingMin(11, {})).toBeUndefined();
      expect(missingMax(9, {})).toBeUndefined();
      expect(missingBoth(1234, {})).toBeUndefined();
    });

    it('should return a validation error message is not inside the segment', () => {
      expect(inSegment(9, {})).toBeDefined();
      expect(inSegment(9, {})).toMatchSnapshot();
      expect(inSegment(21, {})).toBeDefined();
      expect(inSegment(21, {})).toMatchSnapshot();
      expect(inSegment('9', {})).toBeDefined();
      expect(inSegment('9', {})).toMatchSnapshot();
      expect(inSegment('21', {})).toBeDefined();
      expect(inSegment('21', {})).toMatchSnapshot();
      expect(inSegment(new BigNumber(9), {})).toBeDefined();
      expect(inSegment(new BigNumber(9), {})).toMatchSnapshot();
      expect(inSegment(new BigNumber(21), {})).toBeDefined();
      expect(inSegment(new BigNumber(21), {})).toMatchSnapshot();
    });

    it('should throw if range values or input supplied are not treatable as numbers', () => {
      const fail = (value: Ord) => () => {
        const failidator = inSegmentCreator({ min: 'not a number', max: 'somthing' })();
        failidator(value, {});
      };

      expect(fail(15)).toThrowErrorMatchingSnapshot();
      expect(fail('fifteen')).toThrowErrorMatchingSnapshot();
    });

  });
});
