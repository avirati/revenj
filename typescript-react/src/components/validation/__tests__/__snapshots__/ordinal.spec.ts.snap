// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation gtOrdAlphaCreator should return a custom error message on failure when value is not greater than 1`] = `"This makes no sense! Be greater than BBB!"`;

exports[`validation gtOrdAlphaCreator should return a custom error message on failure when value is not greater than 2`] = `"This makes no sense! Be greater than BBB!"`;

exports[`validation gtOrdAlphaCreator should return an error message on value that is less than or equal 1`] = `"Value must be greater than BBB."`;

exports[`validation gtOrdAlphaCreator should return an error message on value that is less than or equal 2`] = `"Value must be greater than BBB."`;

exports[`validation gtOrdCreator should return a custom error message on failure when value is not greater than 1`] = `"This quick brown fox can only jump over more than 10 lazy dogs."`;

exports[`validation gtOrdCreator should return a custom error message on failure when value is not greater than 2`] = `"This quick brown fox can only jump over more than 10 lazy dogs."`;

exports[`validation gtOrdCreator should return a custom error message on failure when value is not greater than 3`] = `"This makes no sense! Be greater than 100!"`;

exports[`validation gtOrdCreator should return a custom error message on failure when value is not greater than 4`] = `"This makes no sense! Be greater than 100!"`;

exports[`validation gtOrdCreator should return an error message on value that is less than or equal 1`] = `"Value must be greater than 10."`;

exports[`validation gtOrdCreator should return an error message on value that is less than or equal 2`] = `"Value must be greater than 10."`;

exports[`validation gtOrdCreator should return an error message on value that is less than or equal 3`] = `"Value must be greater than 100."`;

exports[`validation gtOrdCreator should return an error message on value that is less than or equal 4`] = `"Value must be greater than 100."`;

exports[`validation gteOrdAlphaCreator should return a custom error message on failure when value is not greater than or equal 1`] = `"This makes no sense! Be greater than or equal to BBB!"`;

exports[`validation gteOrdAlphaCreator should return an error message on value that is less than 1`] = `"Value must be greater than or equal to BBB"`;

exports[`validation gteOrdCreator should return a custom error message on failure when value is not greater than or equal 1`] = `"This quick brown fox can only jump over 10 or more lazy dogs."`;

exports[`validation gteOrdCreator should return a custom error message on failure when value is not greater than or equal 2`] = `"This makes no sense! Be greater than or equal to 100!"`;

exports[`validation gteOrdCreator should return an error message on value that is less than 1`] = `"Value must be greater than or equal to 10"`;

exports[`validation gteOrdCreator should return an error message on value that is less than 2`] = `"Value must be greater than or equal to 100"`;

exports[`validation inInterval should return a validation error message is not inside the interval 1`] = `"Value is not in valid range <10, 20>"`;

exports[`validation inInterval should return a validation error message is not inside the interval 2`] = `"Value is not in valid range <10, 20>"`;

exports[`validation inInterval should return a validation error message is not inside the interval 3`] = `"Value is not in valid range <10, 20>"`;

exports[`validation inInterval should return a validation error message is not inside the interval 4`] = `"Value is not in valid range <10, 20>"`;

exports[`validation inInterval should return a validation error message is not inside the interval 5`] = `"Value is not in valid range <10, 20>"`;

exports[`validation inInterval should return a validation error message is not inside the interval 6`] = `"Value is not in valid range <10, 20>"`;

exports[`validation inInterval should throw if range values or input supplied are not treatable as numbers 1`] = `"not a number could not be parsed!"`;

exports[`validation inInterval should throw if range values or input supplied are not treatable as numbers 2`] = `"fifteen could not be parsed!"`;

exports[`validation inSegment should return a validation error message is not inside the segment 1`] = `"Value is not in valid range [10, 20]"`;

exports[`validation inSegment should return a validation error message is not inside the segment 2`] = `"Value is not in valid range [10, 20]"`;

exports[`validation inSegment should return a validation error message is not inside the segment 3`] = `"Value is not in valid range [10, 20]"`;

exports[`validation inSegment should return a validation error message is not inside the segment 4`] = `"Value is not in valid range [10, 20]"`;

exports[`validation inSegment should return a validation error message is not inside the segment 5`] = `"Value is not in valid range [10, 20]"`;

exports[`validation inSegment should return a validation error message is not inside the segment 6`] = `"Value is not in valid range [10, 20]"`;

exports[`validation inSegment should throw if range values or input supplied are not treatable as numbers 1`] = `"not a number could not be parsed!"`;

exports[`validation inSegment should throw if range values or input supplied are not treatable as numbers 2`] = `"fifteen could not be parsed!"`;

exports[`validation ltOrdAlphaCreator should return a custom error message on failure when value is not less than 1`] = `"This makes no sense! Be less than BBB!"`;

exports[`validation ltOrdAlphaCreator should return a custom error message on failure when value is not less than 2`] = `"This makes no sense! Be less than BBB!"`;

exports[`validation ltOrdAlphaCreator should return an error message on value that is greater than or equal 1`] = `"Value must be less than BBB"`;

exports[`validation ltOrdAlphaCreator should return an error message on value that is greater than or equal 2`] = `"Value must be less than BBB"`;

exports[`validation ltOrdCreator should return a custom error message on failure when value is not less than 1`] = `"This quick brown fox can only jump under 10 lazy dogs."`;

exports[`validation ltOrdCreator should return a custom error message on failure when value is not less than 2`] = `"This quick brown fox can only jump under 10 lazy dogs."`;

exports[`validation ltOrdCreator should return a custom error message on failure when value is not less than 3`] = `"This makes no sense! Be less than 100!"`;

exports[`validation ltOrdCreator should return a custom error message on failure when value is not less than 4`] = `"This makes no sense! Be less than 100!"`;

exports[`validation ltOrdCreator should return an error message on value that is greater than or equal 1`] = `"Value must be less than 10"`;

exports[`validation ltOrdCreator should return an error message on value that is greater than or equal 2`] = `"Value must be less than 10"`;

exports[`validation ltOrdCreator should return an error message on value that is greater than or equal 3`] = `"Value must be less than 100"`;

exports[`validation ltOrdCreator should return an error message on value that is greater than or equal 4`] = `"Value must be less than 100"`;

exports[`validation lteOrdAlphaCreator should return a custom error message on failure when value is not less than or equal 1`] = `"This makes no sense! Be less than or equal to BBB!"`;

exports[`validation lteOrdAlphaCreator should return an error message on value that is not less than or equal 1`] = `"Value must be less than or equal to BBB"`;

exports[`validation lteOrdCreator should return a custom error message on failure when value is not less than or equal 1`] = `"This quick brown fox can only jump under 10 or less lazy dogs."`;

exports[`validation lteOrdCreator should return a custom error message on failure when value is not less than or equal 2`] = `"This makes no sense! Be less than or equal to 100!"`;

exports[`validation lteOrdCreator should return an error message on value that is not less than or equal 1`] = `"Value must be less than or equal to 10"`;

exports[`validation lteOrdCreator should return an error message on value that is not less than or equal 2`] = `"Value must be less than or equal to 100"`;
